misc-utils/lslocks.c

```diff
@@ -458,16 +487,73 @@ static void add_scols_line(struct libscols_table *table, struct lock *l, struct
 		if (str && scols_line_refer_data(line, i, str))
 			err(EXIT_FAILURE, _("failed to add output data"));
 	}
+
+	return line;
 }
 
-static int show_locks(struct list_head *locks)
+static struct libscols_column *add_column(struct libscols_table *table, const struct colinfo *col)
 {
-	int rc = 0;
-	size_t i;
-	struct list_head *p, *pnext;
-	struct libscols_table *table;
+	struct libscols_column *cl = scols_table_new_column(table, col->name, col->whint, col->flags);
+	if (!cl)
+		err(EXIT_FAILURE, _("failed to allocate output column"));
+
+	{
+		/* Even if --json/-J is not given, the filter engine requires
+		 * a type must be set to the column. */
+		int json_type = -1;
+
+		if (bytes)
+			json_type = col->json_type_in_bytes_mode;
+		if (json_type == -1)
+			json_type = col->json_type;
+		scols_column_set_json_type(cl, json_type);
```

Comment:
`scols_column_set_json_type(cl, bytes ? col->json_type_in_bytes_mode : col->json_type);`

Reply:
There is one more condition at line 505; if col->json_type_in_bytes_mode is -1, we use col->json_type as a fallback value in the bytes mode.
So using if/else may be better than nested ?: operators.